src/main/java/edu/project/Context.java

```diff
@@ -1,8 +1,25 @@
 package edu.project;
 
-/**
- * Acts as the service locator for all the services in the project
- */
+import edu.project.yourpackage.WeatherApi;
+import edu.project.yourpackage.WeatherCurrentData;
+import edu.project.yourpackage.WeatherForecast5Data;
+import java.io.IOException;  // <-- Make sure this import is present
+import org.jxmapviewer.viewer.GeoPosition;
+
 public class Context {
-	public final Store store = new Store();
+    public Store store;
+    private WeatherApi weatherApi;
+
+    public Context() {
+        this.store = new Store();
+        this.weatherApi = new WeatherApi();
+    }
+
+    public WeatherCurrentData getCurrentWeather(String city) throws IOException {
+        return weatherApi.getCurrentWeather(city);
+    }
+
+    public WeatherForecast5Data getWeatherForecast(String city) throws IOException {
+        return weatherApi.getWeatherForecast(city);
+    }
```

Comment:
The Context class' only purpose is for holding all the services of the app in one location (basically a service locator), we can remove the `getCurrentWeather` & `getWeatherForecast` functions since they're redundant and they can be accessed through the `weatherApi` member. 

We can modify `weatherApi`'s accessibility to be `public final` so that it won't be replaced by outsiders.